import "CommonTopology.proto";

package TopologyModule_Proto;

option java_package = "zsdn.protocol";
option java_outer_classname = "TopologyModuleProtocol";

// All From-messages the TopologyModule will publish

message From {

    oneof FromMsg {
        TopologyChangedEvent topology_changed_event = 1;
    }

    message TopologyChangedEvent{
       required common.topology.Topology topology = 1;
    }

}

// All Requests the TopologyModule will handle

message Request  {

    oneof RequestMsg {
        GetTopologyRequest get_topology_request = 1;
    }

    // Reply submessage definitions

    message GetTopologyRequest {}

}

// All replies the TopologyModule can send

message Reply {

    oneof ReplyMsg {
        GetTopologyReply get_topology_reply = 1;
    }

    // Reply submessage definitions

    message GetTopologyReply {
        required common.topology.Topology topology = 1;
    }
}